internal/core/axis.cpp

```diff
@@ -35,8 +35,11 @@ int Axis::dimension() const noexcept(true) {
     return this->m_dimension;
 }
 
-float Axis::stepsize() const noexcept (true) {
-    return (this->max() - this->min()) / (this->nsamples() - 1);
+float Axis::stepsize() const noexcept(true) {
+    if (this->nsamples() == 1)
```

Comment:
Surround if and else with {} for future safety? Annoying bugs when you just want to add some debugging line into the ifs, don't pay attention to lack of safety brackets and have to figure out why everything broke :smile: